src/a2m-v2.h

```diff
@@ -0,0 +1,850 @@
+    fx_FSlideUp          = 0x01,
+    fx_FSlideDown        = 0x02,
+    fx_FSlideUpFine      = 0x03,
+    fx_FSlideDownFine    = 0x04,
+    fx_TonePortamento    = 0x05,
+    fx_TPortamVolSlide   = 0x06,
+    fx_Vibrato           = 0x07,
+    fx_VibratoVolSlide   = 0x08,
+    fx_SetOpIntensity    = 0x09,
+    fx_SetInsVolume      = 0x0a,
+    fx_PatternBreak      = 0x0b,
+    fx_PatternJump       = 0x0c,
+    fx_SetTempo          = 0x0d,
+    fx_SetTimer          = 0x0e,
+    fx_Extended          = 0x0f,
+    fx_ex_DefAMdepth     = 0x00,
+    fx_ex_DefVibDepth    = 0x01,
+    fx_ex_DefWaveform    = 0x02,
+    fx_ex_ManSlideUp     = 0x03,
+    fx_ex_ManSlideDown   = 0x04,
+    fx_ex_VSlideUp       = 0x05,
+    fx_ex_VSlideDown     = 0x06,
+    fx_ex_VSlideUpFine   = 0x07,
+    fx_ex_VSlideDownFine = 0x08,
+    fx_ex_RetrigNote     = 0x09,
+    fx_ex_SetAttckRate   = 0x0a,
+    fx_ex_SetDecayRate   = 0x0b,
+    fx_ex_SetSustnLevel  = 0x0c,
+    fx_ex_SetReleaseRate = 0x0d,
+    fx_ex_SetFeedback    = 0x0e,
+    fx_ex_ExtendedCmd    = 0x0f
+};
+
+/* Data for importing A2M format */
+typedef struct {
+    tFM_INST_DATA fm;
+    uint8_t panning;
+    int8_t  fine_tune;
+} tINSTR_DATA_V1_8;
+
+C_ASSERT(sizeof(tINSTR_DATA_V1_8) == 13);
+
+typedef struct {
+    char songname[43];
+    char composer[43];
+    char instr_names[250][33];
+    tINSTR_DATA_V1_8 instr_data[250];
+    uint8_t pattern_order[128];
+    uint8_t tempo;
+    uint8_t speed;
+    uint8_t common_flag; // A2M_SONGDATA_V5678
+} A2M_SONGDATA_V1_8;
+
+C_ASSERT(sizeof(A2M_SONGDATA_V1_8) == 11717);
+
+typedef struct {
+    uint8_t num_4op;
+    uint8_t idx_4op[128];
+} tINS_4OP_FLAGS;
+
+typedef uint8_t tRESERVED[1024];
+
+typedef struct {
+    uint8_t rows_per_beat;
+    int8_t tempo_finetune[2]; // int16_t
+} tBPM_DATA;
+
+typedef struct {
+    char songname[43];
+    char composer[43];
+    char instr_names[255][43];
+    tINSTR_DATA instr_data[255];
+    tFMREG_TABLE fmreg_table[255];
+    tARPVIB_TABLE arpvib_table[255];
+    uint8_t pattern_order[128];
+    uint8_t tempo;
+    uint8_t speed;
+    uint8_t common_flag;
+    uint8_t patt_len[2];           // uint16_t
+    uint8_t nm_tracks;
+    uint8_t macro_speedup[2];      // uint16_t
+    uint8_t flag_4op;              // A2M_SONGDATA_V10
+    uint8_t lock_flags[20];        // A2M_SONGDATA_V10
+    char pattern_names[128][43];   // A2M_SONGDATA_V11
+    bool dis_fmreg_col[255][28];   // A2M_SONGDATA_V11
+    tINS_4OP_FLAGS ins_4op_flags;  // A2M_SONGDATA_V12_13
+    tRESERVED reserved_data;       // A2M_SONGDATA_V12_13
+    tBPM_DATA bpm_data;            // A2M_SONGDATA_V14
+} A2M_SONGDATA_V9_14;
+
+C_ASSERT(sizeof(A2M_SONGDATA_V9_14) == 1138338);
+
+/* Player data */
+
+typedef struct {
+    tINSTR_DATA instr_data;
+    uint8_t vibrato;
+    uint8_t arpeggio;
+    tFMREG_TABLE *fmreg;
+    uint32_t dis_fmreg_cols;
+} tINSTR_DATA_EXT;
+
+C_ASSERT(sizeof(tINSTR_DATA_EXT) == 20 + sizeof(tFMREG_TABLE *));
```

Comment:
This fails in my machine when compiled separately from this project: `sizeof(tINSTR_DATA_EXT)` is 32 instead of 28. I had to instruct this to be packed together without padding with this attribute.

```c++
typedef struct {
    tINSTR_DATA instr_data;
    uint8_t vibrato;
    uint8_t arpeggio;
    tFMREG_TABLE *fmreg;
    uint32_t dis_fmreg_cols;
} __attribute__((packed))  tINSTR_DATA_EXT;
```

Unless there is a compilation flag here that does this automatically for all structs, there may be a need to add this attribute in other structs as well.